docs/modules/ROOT/pages/design/cloudevents.adoc

```diff
@@ -0,0 +1,125 @@
+= Cloud Event Delivery Pipeline
+== Overview
+
+KubeArchive uses Cloud Events that are generated by a Knative Eventing ApiServerSource to discover when resources on
+the kubernetes cluster have changed. These Cloud Events contain the json representation of the Kubernetes resource that
+has changed. The KubeArchive Sink (sink) receives these messages and determines, using CEL expression from the
+KubeArchiveConfig, if the data contained in the Cloud Event should be written to the database and potentially deleted
+from the cluster. While the sink is processing a Cloud Event, it is possible at many points for an error to occur,
+especially when the sink has to make a request of the network, particularly, to the Kubernetes API or the database. This
+page proposes how KubeArchive set up its Cloud Event delivery pipeline and how it should consume Cloud Events to achieve
+the following goals:
+
+* KubeArchive receives all Cloud Events from the Knative ApiServerSource
```

Comment:
I don't think this should be a goal, as its stated that KubeArchive already receives CloudEvents from the Knative Eventing ApiServerSource.

Reply:
I removed it from the list of goals